src/python/grpcio/grpc/_interceptor.py

```diff
@@ -142,23 +194,45 @@ def __init__(self, thunk, method, interceptor):
         self._interceptor = interceptor
 
     def __call__(self, request, timeout=None, metadata=None, credentials=None):
-        call_future = self.future(
+        response, ignored_call = self._with_call(
```

Comment:
The whole rest of the gRPC Python codebase maintains definitions-before-uses order; why break that now?